src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,59 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Optional findOne(@PathVariable Long id) throws BookNotFoundException {
+        return userRepository.findById(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
+        return userRepository.save(user);
+    }
+
+    @PostMapping("/addBook/{username}")
+    public User addBookToCollection(@RequestBody Book book, @PathVariable String username) {
+        User user_requested = userRepository.findByUserName(username);
```

Comment:
I'm using `findByUserName` method instead of `findById` because a previous trello card asked me to create that method, so I thought I should use it, given that it was defined previously.